app/assets/javascripts/discourse/app/components/welcome-topic-banner.js

```diff
@@ -0,0 +1,25 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { action } from "@ember/object";
+import { getOwner } from "discourse-common/lib/get-owner";
+import Topic from "discourse/models/topic";
+import Composer from "discourse/models/composer";
+
+export default class WelcomeTopicBanner extends GlimmerComponent {
+  @action
+  editWelcomeTopic() {
+    const topicController = getOwner(this).lookup("controller:topic");
+
+    Topic.find(this.siteSettings.welcome_topic_id, {}).then((topic) => {
```

Comment:
This is the core of this feature but it doesn't look like it's tested. Any chances you can add some?